Realm/RLMRealm.h

```diff
@@ -333,116 +341,126 @@ typedef void (^RLMNotificationBlock)(NSString *notification, RLMRealm *realm);
+ @param object  The object to be added to this Realm.
  */
 - (void)addObject:(RLMObject *)object;
 
 /**
- Adds objects in the given array to be persisted in this Realm.
+ Adds all the objects in a collection to the Realm.
 
- This is the equivalent of `addObject:` except for an array of objects.
+ This is the equivalent of calling `addObject:` for every object in a collection.
 
- @warning This method can only be called during a write transaction.
+ @warning This method may only be called during a write transaction.
 
- @param array   An enumerable object such as NSArray or RLMResults which contains objects to be added to
-                this Realm.
+ @param array   An enumerable object such as `NSArray` or `RLMResults` which contains objects to be added to
+                the Realm.
 
- @see   addObject:
+ @see   `addObject:`
  */
 - (void)addObjects:(id<NSFastEnumeration>)array;
 
 /**
- Adds or updates an object to be persisted in this Realm. The object provided must have a designated
- primary key. If no objects exist in the RLMRealm instance with the same primary key value, the object is
- inserted. Otherwise, the existing object is updated with any changed values.
+ Adds or updates an existing object into the Realm.
+ 
+ The object provided must have a designated primary key. If no objects exist in the Realm 
+ with the same primary key value, the object is inserted. Otherwise, the existing object is
+ updated with any changed values.
 
- As with `addObject:`, the object cannot already be persisted in a different
+ As with `addObject:`, the object cannot already be managed by a different
  Realm. Use `-[RLMObject createOrUpdateInRealm:withValue:]` to copy values to
  a different Realm.
 
- @warning This method can only be called during a write transaction.
+ @warning This method may only be called during a write transaction.
 
- @param object  Object to be added or updated.
+ @param object  The object to be added or updated.
  */
 - (void)addOrUpdateObject:(RLMObject *)object;
 
 /**
- Adds or updates objects in the given array to be persisted in this Realm.
+ Adds or updates all the objects in a collection into the Realm.
 
- This is the equivalent of `addOrUpdateObject:` except for an array of objects.
+ This is the equivalent of calling `addOrUpdateObject:` for every object in a collection.
 
- @warning This method can only be called during a write transaction.
+ @warning This method may only be called during a write transaction.
 
- @param array  `NSArray`, `RLMArray`, or `RLMResults` of `RLMObject`s (or subclasses) to be added to this Realm.
+ @param array  An `NSArray`, `RLMArray`, or `RLMResults` of `RLMObject`s (or subclasses) to be added to the Realm.
 
- @see   addOrUpdateObject:
+ @see   `addOrUpdateObject:`
  */
 - (void)addOrUpdateObjectsFromArray:(id)array;
 
 /**
- Delete an object from this Realm.
+ Deletes an object from the Realm. Once the object is deleted it is considered unmanaged.
```

Comment:
After `deleteObject:` the object is invalidated and you can't do much of anything with it. Saying that it becomes unmanaged could make someone think that it turns back into the state it was before `addObject:` was called on it.